oss_src/unity/extensions/cumulative_aggregates.cpp

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (C) 2015 Dato, Inc.
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the BSD license. See the LICENSE file for details.
+ */
+#include <parallel/pthread_tools.hpp>
+#include <parallel/lambda_omp.hpp>
+#include <unity/lib/toolkit_function_macros.hpp>
+#include <unity/extensions/cumulative_aggregates.hpp>
+
+namespace graphlab {
+namespace cumulative_aggregates {
+
+
+/**
+ * Throw an error if a vector is of unequal length.
+ * \param[in] gl_sarray of type vector 
+ */
+void check_vector_equal_size(const gl_sarray& in) {
+  
+  // Initialize. 
+  DASSERT_TRUE(in.dtype() == flex_type_enum::VECTOR); 
+  size_t n_threads = thread::cpu_count();
+  DASSERT_GE(n_threads, 1);
```

Comment:
I'd rather not assert in this case, as this could happen in the wild. Use std::min